### PR TITLE
React native promise

### PR DESCRIPTION
**Pull request by mhegazy**
Fix for the issue reported in https://github.com/Microsoft/TypeScript/issues/13415

1. remove duplicate declaration of Promise
2. update AsyncStorage definition to match that of the implementation in : https://github.com/facebook/react-native/blob/eba356fb135171b49c2062e9ccac73335e251491/Libraries/Storage/AsyncStorage.js

### CODE REVIEW

**Comment by mhegazy** [comment]
@jemise111 @dannycochran, @rockmacaca, @joshyhargreaves, @jasonhealy, @hktonylee, @horiuchi @cailenmusselman, and @andy-hanson can you review?

File: react-native/index.d.ts [hunk 1]
```diff
@@ -6132,51 +6036,51 @@ declare module "react" {
...
          * Gets all keys known to the app, for all callers, libraries, etc
          */
-        getAllKeys(callback?: (error?: Error, keys?: string[]) => void): Promise<string>
+        getAllKeys(callback?: (error?: Error, keys?: string[]) => void): Promise<string[]>
 
         /**
          * multiGet invokes callback with an array of key-value pair arrays that matches the input format of multiSet
          */
-        multiGet(keys: string[], callback?: (errors?: Error[], result?: string[][]) => void): Promise<string>
+        multiGet(keys: string[], callback?: (errors?: Error[], result?: string[][]) => void): Promise<void>
```
Comment (andy-ms) [inline react-native/index.d.ts]: Callback and return type differ

Comment (mhegazy) [inline react-native/index.d.ts]: seems like this is what the code is doing: https://github.com/facebook/react-native/blob/eba356fb135171b49c2062e9ccac73335e251491/Libraries/Storage/AsyncStorage.js#L340

Comment (andy-ms) [inline react-native/index.d.ts]: Then it gets pushed to `this._getRequests` and resolved here: https://github.com/facebook/react-native/blob/eba356fb135171b49c2062e9ccac73335e251491/Libraries/Storage/AsyncStorage.js#L274
So maybe it's a `[string, string][]`.

Comment (mhegazy) [inline react-native/index.d.ts]: you are right. sorry about that.

**Comment by mhegazy** [comment]
@andy-ms any more comments?

File: react-native/index.d.ts [hunk 2]
```diff
@@ -6132,51 +6036,51 @@ declare module "react" {
...
          * Gets all keys known to the app, for all callers, libraries, etc
          */
-        getAllKeys(callback?: (error?: Error, keys?: string[]) => void): Promise<string>
+        getAllKeys(callback?: (error?: Error, keys?: string[]) => void): Promise<string[]>
 
         /**
          * multiGet invokes callback with an array of key-value pair arrays that matches the input format of multiSet
          */
-        multiGet(keys: string[], callback?: (errors?: Error[], result?: string[][]) => void): Promise<string>
+        multiGet(keys: string[], callback?: (errors?: Error[], result?: string[][]) => void): Promise<[string, string][]>
```
Comment (andy-ms) [inline react-native/index.d.ts]: Types still differ

Comment (mhegazy) [inline react-native/index.d.ts]: third time is a charm ;)

**Review by ghost — APPROVED** [review]

**Pull request merged**